lib/optimizely/event/batch_event_processor.rb

```diff
@@ -90,56 +87,38 @@ def process(user_event)
         return
       end
 
-      @mutex.synchronize do
-        begin
-          @event_queue << user_event
-          @received.signal
-        rescue Exception
-          @logger.log(Logger::WARN, 'Payload not accepted by the queue.')
-          return
-        end
+      begin
+        @event_queue.push(user_event, true)
+      rescue Exception
+        @logger.log(Logger::WARN, 'Payload not accepted by the queue.')
+        return
       end
     end
 
     def stop!
       return unless @started
 
-      @mutex.synchronize do
-        @event_queue << SHUTDOWN_SIGNAL
-        @received.signal
-      end
-
+      @logger.log(Logger::INFO, 'Stopping scheduler.')
+      @event_queue << SHUTDOWN_SIGNAL
+      @thread.join(DEFAULT_TIMEOUT_INTERVAL)
       @started = false
-      @logger.log(Logger::WARN, 'Stopping scheduler.')
-      @thread.exit
     end
 
     private
 
     def run
       loop do
-        if Helpers::DateTimeUtils.create_timestamp > @flushing_interval_deadline
-          @logger.log(
-            Logger::DEBUG,
-            'Deadline exceeded flushing current batch.'
-          )
-          flush_queue!
-        end
+        flush_queue! if Helpers::DateTimeUtils.create_timestamp > @flushing_interval_deadline
```

Comment:
Can you pleae add logs in `flush_queue` while flushing.

Reply:
Done